metadata-ingestion/src/datahub/ingestion/source/unity/proxy.py

```diff
@@ -374,55 +529,139 @@ def list_lineages_by_column(self, table_name: str, column_name: str) -> list:
+        start_time: Optional[datetime] = None,
+        end_time: Optional[datetime] = None,
+    ) -> None:
         if table.schema.catalog.type == CustomCatalogType.HIVE_METASTORE_CATALOG:
             # Lineage is not available for Hive Metastore Tables.
             return None
-        # Lineage endpoint doesn't exists on 2.1 version
-        try:
-            response: dict = self.list_lineages_by_table(
-                table_name=table.ref.qualified_table_name,
-                include_entity_lineage=include_entity_lineage,
-            )
-
-            for item in response.get("upstreams") or []:
-                if "tableInfo" in item:
-                    table_ref = TableReference.create_from_lineage(
-                        item["tableInfo"], table.schema.catalog.metastore
-                    )
-                    if table_ref:
-                        table.upstreams[table_ref] = {}
-                elif "fileInfo" in item:
-                    external_ref = ExternalTableReference.create_from_lineage(
-                        item["fileInfo"]
-                    )
-                    if external_ref:
-                        table.external_upstreams.add(external_ref)
-
-                for notebook in item.get("notebookInfos") or []:
-                    table.upstream_notebooks.add(notebook["notebook_id"])
 
-            for item in response.get("downstreams") or []:
-                for notebook in item.get("notebookInfos") or []:
-                    table.downstream_notebooks.add(notebook["notebook_id"])
+        try:
+            # Use the newer system tables if we have a SQL warehouse, otherwise fall back
+            # to the older (and slower) HTTP API.
+            if self.warehouse_id:
+                self._process_system_table_lineage(table, start_time, end_time)
+            else:
+                self._process_http_api_lineage(table, include_entity_lineage)
         except Exception as e:
             logger.warning(
                 f"Error getting lineage on table {table.ref}: {e}", exc_info=True
             )
 
+    def _process_system_table_lineage(
+        self,
+        table: Table,
+        start_time: Optional[datetime] = None,
+        end_time: Optional[datetime] = None,
+    ) -> None:
+        """Process table lineage using system.access.table_lineage table."""
+        catalog_lineage = self.get_catalog_table_lineage(
+            table.ref.catalog, start_time, end_time
+        )
+        table_full_name = table.ref.qualified_table_name
+
+        lineage_info = catalog_lineage.get(table_full_name, TableLineageInfo())
+
+        # Process table upstreams
+        for upstream in lineage_info.upstreams:
+            upstream_table_name = upstream.table_name
+            # Parse catalog.schema.table format
+            parts = upstream_table_name.split(".")
+            if len(parts) == 3:
+                catalog_name, schema_name, table_name = parts[0], parts[1], parts[2]
```

Comment:
This breaks if a catalog, schema, or table name contain `.`. The table lineage table has `source_table_catalog`, `source_table_schema`, `source_table_name` fields that can be used directly.

Reply:
Thanks for raising, I kept it like this because in Unity Catalog `.` is not allowed in object names -> https://docs.databricks.com/aws/en/sql/language-manual/sql-ref-names